datadog-ktor/src/commonMain/kotlin/Configuration.kt

```diff
@@ -0,0 +1,19 @@
+package com.juul.datadog.ktor
+
+import com.juul.datadog.Site
+
+public data class DatadogKtorConfiguration(
+    public val rest: RestConfiguration,
+    public val log: LogConfiguration = LogConfiguration(),
+)
+
+public data class RestConfiguration(
+    public val apiKey: String,
+    public val site: Site,
+)
+
+public data class LogConfiguration(
+    public val source: String? = null,
+    public val host: String? = null,
+    public val service: String? = null,
+)
```

Comment:
nit: could just go with `Configuration`
If it conflicts, then import alias. It is in a separate package from the other, and not often we'll use both at the same time.

Also, I personally like nesting to keep things organized / tidy. 🤷 

```suggestion
public data class Configuration(
    public val rest: Rest,
    public val log: Log = Log(),
) {

    public data class Rest(
        public val apiKey: String,
        public val site: Site,
    )

    public data class Log(
        public val source: String? = null,
        public val host: String? = null,
        public val service: String? = null,
    )
}

```